test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -220,6 +227,7 @@
     private Function<Client, Client> clientWrapper;
 
     public InternalTestCluster(String nodeMode, long clusterSeed, Path baseDir,
+                               int numMasterNodes,
```

Comment:
I agree with you that this is not ideal. I don't see however how to make it better. I currently the same model we have for client/data nodes where -1 is auto, 0 is non and an explicit number is what it is.  I'm open for a better suggestion but w.r.t booleans - I think that will just add more confusion (for example, what does  dedicateMasterNodes==false, numMasterNodes=4 mean?).